specs/online_order_spec.rb

```diff
@@ -1,90 +1,148 @@
+
+      onlineorder.total.must_equal expected_total
     end
 
     it "Doesn't add a shipping fee if there are no products" do
-      # TODO: Your test code here!
+        onlineorder = Grocery::OnlineOrder.new(12, {}, 987, :fulfilled)
+
+        expected_total = 0
+
+        onlineorder.total.must_equal expected_total
     end
   end
 
   describe "#add_product" do
     it "Does not permit action for processing, shipped or completed statuses" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      before_count = products.count
+      order = Grocery::OnlineOrder.new(1337, products, 485, :shipped)
+
+      order.add_product("salad", 4.25)
+      expected_count = before_count
+      order.products.count.must_equal expected_count
+
     end
 
     it "Permits action for pending and paid satuses" do
-      # TODO: Your test code here!
+      products = { "banana" => 1.99, "cracker" => 3.00 }
+      before_count = products.count
+      order = Grocery::OnlineOrder.new(1337, products, 485, :paid)
+
+      order.add_product("salad", 4.25)
+      expected_count = before_count + 1
+      order.products.count.must_equal expected_count
     end
   end
 
-  describe "OnlineOrder.all" do
+   describe "OnlineOrder.all" do
     it "Returns an array of all online orders" do
-      # TODO: Your test code here!
+        Grocery::OnlineOrder.all.must_be_kind_of Array
+
+        oorder_list = Grocery::OnlineOrder.all
+        oorder_list.each do |i|
+          i.must_be_instance_of Grocery::OnlineOrder
+        end
     end
 
     it "Returns accurate information about the first online order" do
-      # TODO: Your test code here!
+      first_order = Grocery::OnlineOrder.all[0]
+      first_order.id.must_equal "1"
+      first_order.products.must_be_instance_of Hash
+      first_order.products.must_equal({"Lobster" => "17.18", "Annatto seed" => "58.38", "Camomile" => "83.21"})
+      first_order.customer.must_equal "25"
+      first_order.status.must_equal :complete
     end
 
     it "Returns accurate information about the last online order" do
-      # TODO: Your test code here!
+      last_order = Grocery::OnlineOrder.all.last
+      last_order.id.must_equal "100"
+      last_order.products.must_be_instance_of Hash
+      last_order.products.must_equal({"Amaranth" => "83.81", "Smoked Trout" => "70.6", "Cheddar" => "5.63"})
+      last_order.customer.must_equal "20"
+      last_order.status.must_equal :pending
     end
   end
 
   describe "OnlineOrder.find" do
     it "Will find an online order from the CSV" do
-      # TODO: Your test code here!
+      Grocery::OnlineOrder.find("1").must_be_instance_of Grocery::OnlineOrder
+
+      Grocery::OnlineOrder.find("1").products.must_equal ({"Lobster" => "17.18", "Annatto seed" => "58.38", "Camomile" => "83.21"})
+
+      Grocery::OnlineOrder.find("1").customer.must_equal "25"
+      Grocery::OnlineOrder.find("1").status.must_equal :complete
```

Comment:
since this test finds the same order a bunch of times, you probably could just assign a local variable like you do in a lot of the other tests:
```ruby
order = Grocery::OnlineOrder.find("1")
order.must_be_instance_of Grocery::OnlineOrder
order.products.must_equal ...
```